src/Stream.php

```diff
@@ -370,9 +370,14 @@ private function setStream($stream, string $mode = 'r') : void
      */
     private function isValidStreamResourceType($resource): bool
     {
-        return (
-            is_resource($resource) &&
-            in_array(get_resource_type($resource), self::ALLOWED_STREAM_RESOURCE_TYPES, true)
-        );
+        if (is_resource($resource)) {
+            return in_array(get_resource_type($resource), self::ALLOWED_STREAM_RESOURCE_TYPES, true);
+        }
+
+        if (PHP_VERSION_ID >= 80000 && $resource instanceof \GdImage) {
```

Comment:
As this is the only change in here, I have a few questions:

- does all of these functions work on that `GdImage` object?
  - `fclose`
  - `stream_get_contents`
  - `fread`
  - `stream_get_meta_data`

If the answer for any of these functions is "no", we need to find a completely new way to make the `Stream`-implementation compatible with the new `GdImage` object as we are using `fclose` in `Stream::close`, `stream_get_contents` in `Stream::getContents` and `stream_get_metadata` in some of the other methods...

Reply:
I checked locally on PHP 7.3. The following test cases

```
    public function testIsReableWithGdResource()
    {
        $resource = imagecreate(1, 1);
        $stream   = new Stream($resource);
        $this->assertTrue($stream->isReadable());
    }

    public function testCloseWithGdResource()
    {
        $resource = imagecreate(1, 1);
        $stream   = new Stream($resource);
        $this->assertNull($stream->close());
    }

```
results in 
```
1) LaminasTest\Diactoros\StreamTest::testIsReableWithGdResource
stream_get_meta_data(): supplied resource is not a valid stream resource

2) LaminasTest\Diactoros\StreamTest::testCloseWithGdResource
fclose(): supplied resource is not a valid stream resource
```

So the existing code is already broken and doesn't work with image resources.

Reply:
So if you do want to support using image resources it should be sorted out separately as fixing existing broken functionality is outside the scope of this PR.

Reply:
You are right, I just wanted to verify that we have this covered.
I will try to keep in mind that I create an Issue regarding this.